sdk/core/core-rest-pipeline/test/bearerTokenAuthenticationPolicy.spec.ts

```diff
@@ -399,3 +699,163 @@ class MockRefreshAzureCredential implements TokenCredential {
     };
   }
 }
+interface Challenge {
+  testName: string;
+  challenge: string;
+  expectedResponseCode: number;
+  expectedClaims: string | null;
+  encodedClaims: string | null;
```

Comment:
I don't see this read anywhere. Naively, it seems unnecessary because any claims challenge would by definition contain encoded claims, and that form isn't interesting apart from the challenge because the policy passes decoded claims to the credential

Reply:
Yes you're right. If the expected claims match, that already means we have parsed the correct encoded claims. Updated